pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -86,6 +86,9 @@ type VitessShardSpec struct {
 
 	// TopologyReconciliation is inherited from the parent's VitessClusterSpec.
 	TopologyReconciliation *TopoReconcileConfig `json:"topologyReconciliation,omitempty"`
+
+	// InitReplication is inherited from the parent's VitessClusterSpec.
```

Comment:
If it's inherited from the parent's VitessClusterSpec then we should not be running defaulting code in `DefaultVitessShard()`. Defaulting should be up to the `DefaultVitessCluster()` and the value should simply be copied down.

Reply:
Done - Just updated it to copy the value down. And makes sense on the Defaulting code side, we are just setting one value and passing it down.

Reply:
Cool, I'm happy with that part so resolving this conversation so it doesn't clutter up space.

Reply:
Since we're adding a field that's default-on, I think we do need to run defaulting even for inherited fields. Come to think of it, I might have messed this up too in my last PR. Let me go take another look.

Reply:
Oh looks like I did remember to add this in my other PR:

https://github.com/planetscale/vitess-operator/pull/58/files#diff-8d96129ffdb514abe933c2a0416b261bR150

We'll need to do something similar here, to make sure this new vitessshardreplication controller code will see an existing VitessShard object (made by the old vitesscluster controller) as having these default-on fields on.

Reply:
Done, added a line to materialize defaults and put the default function back in vitessshard_defaults.go